### DISCUSSION TITLE
Question about view of bills

### DISCUSSION THREAD

**Discussion by devfaz** [firefly-iii/firefly-iii#7655] [Firefly III]
https://github.com/firefly-iii/firefly-iii/issues/7615#issuecomment-1590479023_

Don't get what you want to say.

If I switch the date to any other month, the bill is not marked as paid in this month.

Also I would not expect to see 2 dates/links?!

What do I miss?

**Comment by JC5** [comment]
> If I switch the date to any other month, the bill is not marked as paid in this month.

Which month is "this" month?

> Also I would not expect to see 2 dates/links?!

You will always see ALL linked transactions.

**Reply by devfaz** [reply]
ok, will try to explain it with some pictures.

The main questions: 

* Why does I see a payment with the date "28th of Feb" in the bills-view, if I selected the month "1st Mar - 31st Mar"
![Bildschirmfoto_2023-06-17_12-13-21](https://github.com/firefly-iii/firefly-iii/assets/4060372/2de599aa-5072-4458-b3b6-246f79319e33)

* You wrote in the bug-report, that the view is "always the same and doesnt take the selected date into account". That cant be true, as I only see the payment of "17th Jan" if I select the "1st Jan - 31st Jan" in the top-date-selector
![Bildschirmfoto_2023-06-17_12-14-35](https://github.com/firefly-iii/firefly-iii/assets/4060372/a5945b20-47d9-49d2-a03d-c0421993837c)

Sadly the demo-site seems to be reset quite often, here steps to re-add the data:

* Create an liability
![Bildschirmfoto_2023-06-17_12-20-09](https://github.com/firefly-iii/firefly-iii/assets/4060372/5b6db3e5-5c6d-4a71-b8f6-6189dbedeb21)

* Create a (monthly) bill
![Bildschirmfoto_2023-06-17_12-19-45](https://github.com/firefly-iii/firefly-iii/assets/4060372/63865a54-25b4-4447-820b-c05df2cdd385)

* Create transactions

**Reply by JC5** [reply]
I see what you mean, thanks for these pictures. I believe it has to do with the original fix for https://github.com/firefly-iii/firefly-iii/issues/7446, but I'm not sure yet how to fix this *and* the original bug.

**Reply by devfaz** [reply]
Im not 100% sure, but I think the issue is older than the fix for #7446 - I will try to setup an ealier version and reproduce the issue

**Reply by JC5** [reply]
I pushed a fix to develop, let me know :+1:

**Reply by devfaz** [reply]
Perfect, looks great - thanks a lot!

**Reply by devfaz** [reply]
Arg, sadly i just got a new display issue

![Bildschirmfoto_2023-07-17_10-49-57](https://github.com/firefly-iii/firefly-iii/assets/4060372/662fa0f6-5bba-4bf1-ad6f-baf3a3c05135)

the above line is a monthly bill and the view should only display 1st July - 31rd July.

**Reply by devfaz** [reply]
The "/reports/default/allAssetAccounts/20230701/20230731" does not show the bill as "billed", so it seems to be just an display issue in "/bills"

**Reply by JC5** [reply]
Which version are you on?

**Reply by devfaz** [reply]
6.0.17 - renovate is updating my install as soon as you push to dockerhub ^^ :)

**Reply by devfaz** [reply]
Hint: Changing the date of the transaction to the 29th of Jun "fixes" the display issue.

**Reply by JC5** [reply]
Indeed. I believe I have fixed it in the develop branch, and I will push a new release together with some other minor fixes.

**Reply by devfaz** [reply]
Hi, just updated to 6.0.18 - display bug is still there. Just in case

"Bill display would include transactions from the previous period."

should fix this issue here.

**Reply by JC5** [reply]
Nah it should be fixed. I'll try again later.

**Reply by devfaz** [reply]
Hi,

is there anything I could do to help? Should I take a look into the code?

**Reply by JC5** [reply]
That's alright thx. I have time soon

**Comment by devfaz** [comment]
just created https://github.com/firefly-iii/firefly-iii/pull/8026

**Reply by JC5** [reply]
I pushed a potential fix to the develop image, what do you think?

**Reply by devfaz** [reply]
my brain has really hard to work to get all these "->subDay()" ;)

I commited a version which seems to work correctly for me and it works without ->subDay(). But Im quite sure I dont have all the edge-cases in my mind.
